lib/modules/datasource/hexpm-bob/index.ts

```diff
@@ -0,0 +1,122 @@
+import is from '@sindresorhus/is';
+import { logger } from '../../../logger';
+import { ExternalHostError } from '../../../types/errors/external-host-error';
+import { cache } from '../../../util/cache/package/decorator';
+import { HttpError } from '../../../util/http';
+import { Datasource } from '../datasource';
+import type { GetReleasesConfig, ReleaseResult } from '../types';
+import { datasource, defaultRegistryUrl } from './common';
+import type { PackageType } from './types';
+
+export class HexpmBobDatasource extends Datasource {
+  static readonly id = datasource;
+
+  constructor() {
+    super(datasource);
+  }
+
+  override readonly customRegistrySupport = true;
+
+  override readonly defaultRegistryUrls = [defaultRegistryUrl];
+
+  override readonly caching = true;
+
+  @cache({
+    namespace: `datasource-${datasource}`,
+    key: ({ registryUrl, packageName }: GetReleasesConfig) =>
+      `${registryUrl ?? defaultRegistryUrl}:${packageName}`,
+  })
+  async getReleases({
+    registryUrl,
+    packageName,
+  }: GetReleasesConfig): Promise<ReleaseResult | null> {
+    const packageType = this.getPackageType(packageName);
+
+    if (!packageType) {
+      return null;
+    }
+
+    logger.trace(
+      { registryUrl: registryUrl ?? defaultRegistryUrl, packageName },
+      `fetching hex.pm bob ${packageName} release`
+    );
+
+    const url = `${
+      registryUrl ?? defaultRegistryUrl
```

Comment:
```suggestion
      registryUrl!
```

never undefined. It's currently just a typescript type issue